core/templates/pages/blog-dashboard-page/blog-post-editor/blog-post-editor.component.ts

```diff
@@ -0,0 +1,333 @@
+      .then(
+        (editorData: BlogPostEditorData) => {
+          this.blogPostData = editorData.blogPostDict;
+          this.authorProfilePictureUrl = decodeURIComponent((
+            // eslint-disable-next-line max-len
+            editorData.profilePictureDataUrl || this.DEFAULT_PROFILE_PICTURE_URL));
+          this.authorUsername = editorData.username;
+          this.defaultTagsList = editorData.listOfDefaulTags;
+          this.maxAllowedTags = editorData.maxNumOfTags;
+          this.title = this.blogPostData.title;
+          this.dateTimeLastSaved = this.getDateStringInWords(
+            this.blogPostData.lastUpdated);
+          if (this.blogPostData.content.length > 0) {
+            this.contentEditorIsActive = false;
+          }
+          if (this.blogPostData.thumbnailFilename) {
+            this.thumbnailDataUrl = this.assetsBackendApiService
+              .getThumbnailUrlForPreview(
+                AppConstants.ENTITY_TYPE.BLOG_POST, this.blogPostId,
+                this.blogPostData.thumbnailFilename);
+            if (this.windowIsNarrow) {
+              this.blogDashboardPageService.imageUploaderIsNarrow = true;
+            }
+          }
+          if (this.blogPostData.lastUpdated === this.blogPostData.publishedOn) {
+            this.lastChangesWerePublished = true;
+          }
+        }, (errorResponse) => {
+          if (
+            AppConstants.FATAL_ERROR_CODES.indexOf(
+              errorResponse.status) !== -1) {
+            this.alertsService.addWarning('Failed to get blog post data.');
+          }
+        });
+  }
+
+  getDateStringInWords(naiveDateTime: string): string {
+    let datestring = naiveDateTime.substring(0, naiveDateTime.length - 7);
+    return dayjs(
+      datestring, 'MM-DD-YYYY, HH:mm:ss').format('MMMM D, YYYY [at] hh:mm A');
+  }
+
+  updateLocalTitleValue(): void {
+    this.blogPostUpdateService.setBlogPostTitle(
+      this.blogPostData, this.title);
+    this.newChangesAreMade = true;
+  }
+
+  cancelEdit(): void {
+    if (this.blogPostData.content.length > 0) {
+      this.contentEditorIsActive = false;
+    }
+  }
+
+  updateLocalEditedContent($event: string): void {
+    if (this.localEditedContent !== $event) {
+      this.localEditedContent = $event;
+      this.changeDetectorRef.detectChanges();
+    }
+  }
+
+  updateContentValue(): void {
+    this.blogPostUpdateService.setBlogPostContent(
+      this.blogPostData, this.localEditedContent);
+    if (this.blogPostData.content.length > 0) {
+      this.contentEditorIsActive = false;
+    }
+    this.newChangesAreMade = true;
+  }
+
+  saveDraft(): void {
+    let issues = this.blogPostData.validate();
+    if (issues.length === 0) {
+      this.updateBlogPostData(false);
+    } else {
+      this.alertsService.addWarning(
+        'Please fix the errors.'
+      );
+    }
+  }
+
+  headersAreEnabledCallBack(): boolean {
+    return true;
+  }
```

Comment:
Why is this required in a function form?

Reply:
angularjs directives access their passed values by calling them like a function. As, schema based editor is an angularjs directive, we have to pass the headers in form of callback. This is also done for localValue in schema based editor.